state/azure/blobstorage/blobstorage.go

```diff
@@ -91,18 +92,24 @@ func (r *StateStore) Init(metadata state.Metadata) error {
 	p := azblob.NewPipeline(credential, options)
 
 	var containerURL azblob.ContainerURL
+	var URL *url.URL
 	customEndpoint, ok := metadata.Properties[endpointKey]
 	if ok && customEndpoint != "" {
-		URL, parseErr := url.Parse(fmt.Sprintf("%s/%s/%s", customEndpoint, meta.accountName, meta.containerName))
-		if parseErr != nil {
-			return parseErr
+		URL, err = url.Parse(fmt.Sprintf("%s/%s/%s", customEndpoint, meta.accountName, meta.containerName))
+		if err != nil {
+			return err
 		}
 		containerURL = azblob.NewContainerURL(*URL, p)
```

Comment:
Put this line outside if block since it is universal.

Reply:
Pulled out containerURL and error check 
Both code path have now error check